### ISSUE TITLE
CreateContainer Problem on Windows 10 

### ISSUE THREAD

**Issue by AhnMo** [docker/docker#29768]
**Description**
At first I open this issue to https://github.com/docker/for-win/issues/361, but I was advised that it have to open here.

When I try to start container, I got a message ,`Error response from daemon: container [Container Name] encountered an error during CreateContainer: failure in a Windows system call: The handle is invalid. (0x6)`.

**Steps to reproduce the issue:**
I have no idea to reproduce. It seems environments error...? 

**Describe the results you received:**

I installed docker as shown on the Microsoft homepage(https://docs.microsoft.com/ko-kr/virtualization/windowscontainers/quick_start/quick_start_windows_10).

But there is no `docker-1.13.0-dev.zip`, so I just replace it to `docker-1.14.0-dev.zip`.

I just try to run container.

> PS C:\WINDOWS\system32> docker ps -a
CONTAINER ID        IMAGE                  COMMAND             CREATED             STATUS              PORTS
   NAMES
9f41f09e9678        microsoft/nanoserver   "cmd"               2 hours ago         Created
   quizzical_goodall
PS C:\WINDOWS\system32> docker run -it microsoft/nanoserver cmd
c:\program files\docker\docker.exe: Error response from daemon: container 73bc6ac7b17b8f5bf6ff72d8c845606d858a52badcc0a6
6d81713ff0a7113263 encountered an error during CreateContainer: failure in a Windows system call: The handle is invalid.
 (0x6).
PS C:\WINDOWS\system32> docker ps -a
CONTAINER ID        IMAGE                  COMMAND             CREATED             STATUS              PORTS
   NAMES
73bc6ac7b17b        microsoft/nanoserver   "cmd"               7 seconds ago       Created
   adoring_ritchie
9f41f09e9678        microsoft/nanoserver   "cmd"               2 hours ago         Created
   quizzical_goodall
PS C:\WINDOWS\system32>

And `docker events`.

> PS C:\WINDOWS\system32> docker events
2016-12-29T02:50:40.758619200+09:00 container create 73bc6ac7b17b8f5bf6ff72d8c845606d858a52badcc0a66d81713ff0a7113263 (i
mage=microsoft/nanoserver, name=adoring_ritchie)
2016-12-29T02:50:40.759617800+09:00 container attach 73bc6ac7b17b8f5bf6ff72d8c845606d858a52badcc0a66d81713ff0a7113263 (i
mage=microsoft/nanoserver, name=adoring_ritchie)
2016-12-29T02:50:41.244616700+09:00 network connect e4eaa39f1fbe8043361ff93ea0be451f91728b039eb2b92dd0c8a7be3394937d (co
ntainer=73bc6ac7b17b8f5bf6ff72d8c845606d858a52badcc0a66d81713ff0a7113263, name=nat, type=nat)
2016-12-29T02:50:42.575627800+09:00 network disconnect e4eaa39f1fbe8043361ff93ea0be451f91728b039eb2b92dd0c8a7be3394937d
(container=73bc6ac7b17b8f5bf6ff72d8c845606d858a52badcc0a66d81713ff0a7113263, name=nat, type=nat)
[NO MORE]

What can I do to solve this problem? Any suggestions?

**Describe the results you expected:**
Just running container and attach container's cmd shell.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
PS C:\WINDOWS\system32> docker version
Client:
 Version:      1.14.0-dev
 API version:  1.26
 Go version:   go1.7.4
 Git commit:   9fab262
 Built:        Wed Dec 28 13:54:30 2016
 OS/Arch:      windows/amd64

Server:
 Version:      1.14.0-dev
 API version:  1.26 (minimum version 1.24)
 Go version:   go1.7.4
 Git commit:   9fab262
 Built:        Wed Dec 28 13:54:30 2016
 OS/Arch:      windows/amd64
 Experimental: false
PS C:\WINDOWS\system32>
```

**Output of `docker info`:**

```
PS C:\WINDOWS\system32> docker info
Containers: 2
 Running: 0
 Paused: 0
 Stopped: 2
Images: 1
Server Version: 1.14.0-dev
Storage Driver: windowsfilter
 Windows:
Logging Driver: json-file
Plugins:
 Volume: local
 Network: l2bridge l2tunnel nat null overlay transparent
Swarm: inactive
Default Isolation: hyperv
Kernel Version: 10.0 14393 (14393.576.amd64fre.rs1_release_inmarket.161208-2252)
Operating System: Windows 10 Pro
OSType: windows
Architecture: x86_64
CPUs: 4
Total Memory: 3.913 GiB
Name: DESKTOP-GAMESHN
ID: RMXO:BH2E:HT6D:YNNK:QJ5Y:GDL7:G7HK:5HDH:UMAX:E4HS:TMKI:ZO5C
Docker Root Dir: C:\ProgramData\docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
PS C:\WINDOWS\system32>
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

CPU: i5-2430M
RAM: 4.00GB
Model Name: Lenovo Thinkpad E420
OS: Windows 10 Pro (14393.576)

**Comment by cpuguy83** [comment]
ping @jhowardmsft

**Comment by Steoghe** [comment]
same issue as AhnMo

**Comment by alefesta** [comment]
I had the same issues: solved reinstalling docker and the windows image. My environment now:
```
PS C:\> docker version
Client:
 Version:      1.14.0-dev
 API version:  1.26
 Go version:   go1.7.4
 Git commit:   a247bf0
 Built:        Thu Dec 29 21:36:56 2016
 OS/Arch:      windows/amd64

Server:
 Version:      1.14.0-dev
 API version:  1.26 (minimum version 1.24)
 Go version:   go1.7.4
 Git commit:   a247bf0
 Built:        Thu Dec 29 21:36:56 2016
 OS/Arch:      windows/amd64
 Experimental: false
```

**Comment by jhowardmsft** [comment]
Let's try and rule some things out.  First one being networking. Try `docker run --rm --net=none microsoft/nanoserver cmd /s /c echo hello`.

If that works, and fails when you don't include the `--net=none`, can you stop the daemon `net stop docker` and run it manually from an elevated command prompt with debugging enabled `dockerd -D` so we can get the debug logs out easily to give more clues as to what might be going on.

If including `--net=none` still fails, we'll need some event logs out of the system.

**Comment by AhnMo** [comment]
Oh ... I think you've got it right. 

Here is the result.
> PS C:\WINDOWS\system32> docker run --rm --net=none microsoft/nanoserver cmd /s /c echo hello
hello
PS C:\WINDOWS\system32>

And here is `docker events`

> PS C:\WINDOWS\system32> docker events
2017-01-05T00:06:22.583277000+09:00 container create 361d31946376f103e2d21d654c02ef53041d7ea44fb1be71eef1259fa2503322 (i
mage=microsoft/nanoserver, name=stupefied_khorana)
2017-01-05T00:06:22.619277600+09:00 container attach 361d31946376f103e2d21d654c02ef53041d7ea44fb1be71eef1259fa2503322 (i
mage=microsoft/nanoserver, name=stupefied_khorana)
2017-01-05T00:06:22.749271300+09:00 network connect 4ea25f005d5eb078d763917a8549ce19e0fba8dcc0a24754bf3f0fc080135b9a (co
ntainer=361d31946376f103e2d21d654c02ef53041d7ea44fb1be71eef1259fa2503322, name=none, type=null)
2017-01-05T00:07:30.105283200+09:00 container start 361d31946376f103e2d21d654c02ef53041d7ea44fb1be71eef1259fa2503322 (im
age=microsoft/nanoserver, name=stupefied_khorana)
2017-01-05T00:07:33.559279800+09:00 container die 361d31946376f103e2d21d654c02ef53041d7ea44fb1be71eef1259fa2503322 (exit
Code=0, image=microsoft/nanoserver, name=stupefied_khorana)
2017-01-05T00:07:33.814286500+09:00 network disconnect 4ea25f005d5eb078d763917a8549ce19e0fba8dcc0a24754bf3f0fc080135b9a
(container=361d31946376f103e2d21d654c02ef53041d7ea44fb1be71eef1259fa2503322, name=none, type=null)
2017-01-05T00:07:34.412287100+09:00 container destroy 361d31946376f103e2d21d654c02ef53041d7ea44fb1be71eef1259fa2503322 (
image=microsoft/nanoserver, name=stupefied_khorana)

Is there any temporary solution before docker engine update?

**Comment by jhowardmsft** [comment]
@jmesser81 have you got a link to the networking debug/reset script?

**Comment by friism** [comment]
@jhowardmsft @AhnMo it's here: https://github.com/Microsoft/Virtualization-Documentation/tree/live/windows-server-container-tools/CleanupContainerHostNetworking

**Comment by AhnMo** [comment]
Cool!

I just run `CleanupContainerHostNetworking.ps1` with no options, and no problem without `--net=none`.

> PS C:\WINDOWS\system32> docker events
2017-01-05T10:06:08.651252400+09:00 container create d75d72e41777e667d53f1f0d6f426125b4128ba4b6aa8b40c28cf15078ac2d2b (image=microsoft/nanoserver, name=affectionate_agnesi)
2017-01-05T10:06:08.653249800+09:00 container attach d75d72e41777e667d53f1f0d6f426125b4128ba4b6aa8b40c28cf15078ac2d2b (image=microsoft/nanoserver, name=affectionate_agnesi)
2017-01-05T10:06:09.057264300+09:00 network connect 0c17f01dacefc201f29fc4fd244a47da76812fc406b92f5cbc459d1c5757af3d (container=d75d72e41777e667d53f1f0d6f426125b4128ba4b6aa8b40c28cf15078ac2d2b, name=nat, type=nat)
2017-01-05T10:06:19.849246700+09:00 container start d75d72e41777e667d53f1f0d6f426125b4128ba4b6aa8b40c28cf15078ac2d2b (image=microsoft/nanoserver, name=affectionate_agnesi)
2017-01-05T10:06:19.889247300+09:00 container resize d75d72e41777e667d53f1f0d6f426125b4128ba4b6aa8b40c28cf15078ac2d2b (height=50, image=microsoft/nanoserver, name=affectionate_agnesi, width=120)

Thank you, guys!

**Comment by AhnMo** [comment]
@friism I have a question!.. I read `The default location of this log directory is under the current working directory from which the command was run.`. It means if I try to execute in some directory that I'm permitted, this issue not gonna happen?? 

Then how about default log location change to `%PROGRAMDATA%\docker\logs`??

This is quick and dirty solution for someone who do not know in detail just like me.

**Comment by friism** [comment]
@AhnMo I don't know, I recommend opening an issue over here if you're encountering problems running the debug script: https://github.com/Microsoft/Virtualization-Documentation

**Comment by elutsky** [comment]
Happens to me as well, when trying to run multiple containers for same image.

Reproduction steps:
1. Using microsoft/iis oficial image
2. Run first container "_docker run -d --name first microsoft/iis_" - **run successfully**
3. Run second container "_docker run -d --name second microsoft/iis_" - **failed**
4. Stop first container
5. Start second container - ## **started successfully!**
6. Try to start first container - failed.

**Comment by jhowardmsft** [comment]
@elutsky - can you open a new issue - this is quite different. The problem in your case appears to be that `docker start` on an already running container doesn't display an error to the client (even though it fails). I can't repro step 6 though:

```
PS E:\> docker run -d --name first microsoft/iis
9e2f5a2841a52b5e2a6aedcb8f486b4aff64ef07310f58117e9e43eaab098b0d
PS E:\> docker ps -a
CONTAINER ID        IMAGE               COMMAND                   CREATED             STATUS              PORTS               NAMES
9e2f5a2841a5        microsoft/iis       "C:\\ServiceMonitor..."   10 seconds ago      Up 1 second                             first
PS E:\> docker run -d --name second microsoft/iis
0dafbf4c3d2dc027826f17096678395dfaec89565065c76a712a22b8dac22d41
PS E:\> docker ps -a
CONTAINER ID        IMAGE               COMMAND                   CREATED             STATUS              PORTS               NAMES
0dafbf4c3d2d        microsoft/iis       "C:\\ServiceMonitor..."   6 seconds ago       Up 4 seconds                            second
9e2f5a2841a5        microsoft/iis       "C:\\ServiceMonitor..."   31 seconds ago      Up 22 seconds                           first
PS E:\> docker stop first
first
PS E:\> docker ps -a
CONTAINER ID        IMAGE               COMMAND                   CREATED             STATUS                              PORTS               NAMES
0dafbf4c3d2d        microsoft/iis       "C:\\ServiceMonitor..."   17 seconds ago      Up 14 seconds                                           second
9e2f5a2841a5        microsoft/iis       "C:\\ServiceMonitor..."   42 seconds ago      Exited (3221225786) 2 seconds ago                       first
PS E:\> docker start second
second
PS E:\> docker ps -a
CONTAINER ID        IMAGE               COMMAND                   CREATED              STATUS                               PORTS               NAMES
0dafbf4c3d2d        microsoft/iis       "C:\\ServiceMonitor..."   40 seconds ago       Up 37 seconds                                            second
9e2f5a2841a5        microsoft/iis       "C:\\ServiceMonitor..."   About a minute ago   Exited (3221225786) 25 seconds ago                       first
PS E:\> # Daemon log: time="2017-02-15T14:45:14.839769500-08:00" level=error msg="Handler for POST /v1.27/containers/second/start returned error: Container a
lready started". No error was returned to client
PS E:\> docker start first
first
PS E:\> docker ps -a
CONTAINER ID        IMAGE               COMMAND                   CREATED              STATUS              PORTS               NAMES
0dafbf4c3d2d        microsoft/iis       "C:\\ServiceMonitor..."   About a minute ago   Up About a minute                       second
9e2f5a2841a5        microsoft/iis       "C:\\ServiceMonitor..."   2 minutes ago        Up 8 seconds                            first
PS E:\>

```

**Comment by jhowardmsft** [comment]
Note above using docker master, using Windows Server (not Hyper-V) containers on Windows Server 2016 fully patched to the "1B" release - 14393.rs1_release.161220-1747.
```
PS E:\> docker version
Client:
 Version:      1.14.0-dev
 API version:  1.27
 Go version:   go1.7.5
 Git commit:   f18bc25
 Built:        02/15/2017 21:58:59
 OS/Arch:      windows/amd64

Server:
 Version:      1.14.0-dev
 API version:  1.27 (minimum version 1.24)
 Go version:   go1.7.5
 Git commit:   f18bc25
 Built:        02/15/2017 21:58:59
 OS/Arch:      windows/amd64
 Experimental: false
```

**Comment by jhowardmsft** [comment]
And further note - this appears to be the same as on Linux - `docker stop foo` on an already stopped container doesn't give any error at the client.

**Comment by elutsky** [comment]
@jhowardmsft for me - issue solved now.
I am not sure what exactly solved it - update to version 1.13.1 or restart my pc :)
May be will help for other...
And no - it was not "stop stopped container" or "start started container".
I have 2 containers, and did try to run both of them simultaneously. But was able to run only one. Any of them. The second was failed.
